sdk/core/Azure.Core.Experimental/src/RequestOptions.cs

```diff
@@ -30,6 +35,46 @@ public class RequestOptions
         /// <param name="perCall"></param>
         public RequestOptions(Action<HttpMessage> perCall) => PerCallPolicy = new ActionPolicy(perCall);
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="treatAsSuccess"></param>
+        public RequestOptions(params int[] treatAsSuccess) : this(treatAsSuccess, ResponseClassification.Success)
```

Comment:
I think we should rename the parameter to something along the lines of "don't throw on the following status codes". I think treatAsSuccess is overpromising.  Some of these calls will result in failures it's just that we won't throw because we saw some status code.